pp_netlib/gt_prune.py

```diff
@@ -0,0 +1,293 @@
+                            needs_unword = True
+                            if cls_id == None:
+                                cls_id = old_cluster_name
+                            else:
+                                cls_id += "_" + old_cluster_name
+                        # Exact match -> same name as before
+                        elif len(join) == len(ref_only):
+                            assert merge == False # should not have already been part of a merge
+                            cls_id = old_cluster_name
+                            break
+
+            # Report merges
+            if merge:
+                merged_ids = cls_id.split("_")
+                sys.stderr.write("Clusters " + ",".join(merged_ids) + " have merged into " + cls_id + "\n")
+
+        # Otherwise just number sequentially starting from 1
+        else:
+            cls_id = new_cls_idx + 1
+            needs_unword = True
+
+        if write_unwords and needs_unword:
+            unword = next(unword_generator)
+        else:
+            unword = None
+
+        for cluster_member in new_cluster:
+            clustering[cluster_member] = cls_id
+            if unword is not None:
+                cluster_unword[cluster_member] = unword
+
+    # print clustering to file
+    if print_csv:
+        out_filename = out_prefix + "_clusters.csv"
+        with open(out_filename, 'w') as cluster_file:
+            cluster_file.write("Taxon,Cluster\n")
+            if write_unwords:
+                unword_file = open(out_prefix + "_unword_clusters.csv", 'w')
+                unword_file.write("Taxon,Cluster_name\n")
+
+            # sort the clusters by frequency - define a list with a custom sort order
+            # first line gives tuples e.g. (1, 28), (2, 17) - cluster 1 has 28 members, cluster 2 has 17 members
+            # second line takes first element - the cluster IDs sorted by frequency
+            freq_order = sorted(dict(Counter(clustering.values())).items(), key=operator.itemgetter(1), reverse=True)
+            freq_order = [x[0] for x in freq_order]
+
+            # iterate through cluster dictionary sorting by value using above custom sort order
+            for cluster_member, cluster_name in sorted(clustering.items(), key=lambda i:freq_order.index(i[1])):
+                if print_ref or cluster_member not in old_names:
+                    cluster_file.write(",".join((cluster_member, str(cluster_name))) + "\n")
+                if write_unwords and cluster_member in cluster_unword:
+                    unword_file.write(",".join((cluster_member, cluster_unword[cluster_member])) + "\n")
+
+            if write_unwords:
+                unword_file.close()
+
+        if external_cluster_csv is not None:
+            print_external_clusters(new_clusters, external_cluster_csv, out_prefix, old_names, print_ref)
+
+    return clustering 
+
+def gt_get_ref_graph(graph, ref_indices, vertex_labels, type_isolate, backend):
```

Comment:
Please describe what this function does. It prunes the graph given a list of references? What else?

Reply:
Also, I don't think backend is used

Reply:
this function simply returns pruned graph

Reply:
Sorry, unclear – I mean please write a comment/docstring in the code to describe what the function does